### PR TITLE
#1481: removed use of the any type

### LINKED ISSUE #1481: [Refactoring] - Remove `any` type from `src/frontend/src/apis/finance.api.ts`

### PR DESCRIPTION
**Pull request by QW66666**
## Changes

removed usage of `any` type in getAllExpenseTypes function

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1481

### CODE REVIEW

**Comment by ninest** [comment]
Looks good! Can you tick all complete checkboxes in the PR description

**Comment by QW66666** [comment]
> Looks good! Can you tick all complete checkboxes in the PR description

Yeah, I just did it

File: src/frontend/src/apis/finance.api.ts [hunk 1]
```diff
@@ -82,8 +83,10 @@ export const deleteReimbursementRequest = (id: string) => {
  */
 export const getAllExpenseTypes = () => {
   return axios.get(apiUrls.getAllExpenseTypes(), {
-    transformResponse: (data) =>
-      JSON.parse(data).map((expenseType: any) => ({ ...expenseType, id: expenseType.expenseTypeId }))
+    transformResponse: (data) => {
+      const parsedData = JSON.parse(data) as ExpenseType[];
+      return parsedData.map((expenseType) => ({ ...expenseType, id: expenseType.expenseTypeId }));
```
Comment (RChandler234) [inline src/frontend/src/apis/finance.api.ts:88]: Not sure why we even had this code setting the id like this, you can condense the method to just return JSON.parse(data) as ExpenseType[]

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/apis/finance.api.ts (hunk 1 above)
Comment (ninest) [reply src/frontend/src/apis/finance.api.ts]: To be honest I wasn't so sure but this was the previous code

```ts
    transformResponse: (data) =>
      JSON.parse(data).map((expenseType: any) => ({ ...expenseType, id: expenseType.expenseTypeId }))
```

does id have to be set to expenseTypeId?

Comment (QW66666) [reply src/frontend/src/apis/finance.api.ts]: I just changed it

**Review by ninest — APPROVED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**